File: golang/concurrency.go
package main

import (
    "fmt"
    "sync"
    "time"
)

func worker(wg *sync.WaitGroup) {
    defer wg.Done()
    sum := 0
    for i := 0; i < 1000000; i++ {
        sum += i
    }
}

func main() {
    start := time.Now()
    var wg sync.WaitGroup
    for i := 0; i < 100; i++ {
        wg.Add(1)
        go worker(&wg)
    }
    wg.Wait()
    elapsed := time.Since(start)
    fmt.Printf("Time: %v\n", elapsed)
}

File: swift/concurrency.swift
import Foundation

let start = Date()
let group = DispatchGroup()
let queue = DispatchQueue.global(qos: .userInitiated)

for _ in 0..<100 {
    queue.async(group: group) {
        var sum = 0
        for i in 0..<1_000_000 {
            sum += i
        }
    }
}

group.wait()
let elapsed = Date().timeIntervalSince(start)
print("Time: \(elapsed * 1000) ms") // Convert seconds to ms
